{"context": "eslint-plugin-eslint-comments \u306f\u3001ESLint \u30c7\u30a3\u30ec\u30af\u30c6\u30a3\u30d6 \u30b3\u30e1\u30f3\u30c8 (/*eslint-disable*/\u306e\u3088\u3046\u306a\u3084\u3064) \u3092\u4f7f\u3046\u3068\u304d\u306e\u30d9\u30b9\u30c8\u30d7\u30e9\u30af\u30c6\u30a3\u30b9\u3092\u307e\u3068\u3081\u305f ESLint \u306e\u8ffd\u52a0\u30eb\u30fc\u30eb\u96c6\u3067\u3059\u3002\nESLint \u30c7\u30a3\u30ec\u30af\u30c6\u30a3\u30d6 \u30b3\u30e1\u30f3\u30c8\u3092\u4f7f\u3046\u3068\u304d\u306f\u3001\u610f\u56f3\u305b\u305a\u6709\u7528\u306a\u8b66\u544a\u304c\u6291\u5236\u3055\u308c\u3066\u3057\u307e\u3046\u3053\u3068\u3092\u9632\u3050\u305f\u3081\u3001\u53ef\u80fd\u306a\u9650\u308a\u5f71\u97ff\u7bc4\u56f2\u304c\u72ed\u3044\u3088\u3046\u306b\u8a18\u8ff0\u3059\u3079\u304d\u3067\u3059\u3002\n\u3053\u306e\u30d7\u30e9\u30b0\u30a4\u30f3\u306e\u30eb\u30fc\u30eb\u306f\u3001\u305d\u306e\u305f\u3081\u306e\u624b\u52a9\u3051\u3092\u3057\u3066\u304f\u308c\u307e\u3059\u3002\n\n \u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\nESLint \u30d7\u30e9\u30b0\u30a4\u30f3\u3067\u3059\u306e\u3067\u3001npm \u3092\u5229\u7528\u3057\u3066\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u307e\u3059\u3002\nESLint \u3092\u30b0\u30ed\u30fc\u30d0\u30eb (--global) \u306b\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u3066\u3044\u308b\u5834\u5408\u306f\u3001\u540c\u3058\u304f\u30b0\u30ed\u30fc\u30d0\u30eb\u3078\u3002ESLint \u3092\u30ed\u30fc\u30ab\u30eb (--save-dev) \u306b\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u3066\u3044\u308b\u5834\u5408\u306f\u3001\u540c\u3058\u304f\u30ed\u30fc\u30ab\u30eb\u3078\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u3066\u304f\u3060\u3055\u3044\u3002\n\nShell\n$ npm install --save-dev eslint eslint-plugin-node\n\n\n\nNode.js 4.0.0 \u4ee5\u964d\u304c\u5fc5\u8981\u3067\u3059\u3002\nESLint 3.1.0 \u4ee5\u964d\u304c\u5fc5\u8981\u3067\u3059\u3002\n\n\u306a\u304a\u3001\u63a8\u5968\u306f\u30ed\u30fc\u30ab\u30eb \u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3067\u3059\u3002\n\n \u4f7f\u3044\u65b9\n\u8a2d\u5b9a\u30d5\u30a1\u30a4\u30eb (.eslintrc.json\u7b49) \u306b\u30d7\u30e9\u30b0\u30a4\u30f3\u3068\u30eb\u30fc\u30eb\u306e\u8a2d\u5b9a\u3092\u8ffd\u52a0\u3057\u307e\u3059\u3002\n\n\u4f8b\n{\n    \"extends\": [\"eslint:recommended\"],\n    \"plugins\": [\"eslint-comments\"],\n    \"rules\": {\n        \"eslint-comments/disable-enable-pair\": \"error\",\n        \"eslint-comments/no-duplicate-disable\": \"error\",\n        \"eslint-comments/no-unlimited-disable\": \"error\",\n        \"eslint-comments/no-unused-disable\": \"error\",\n        \"eslint-comments/no-unused-enable\": \"error\",\n        \"eslint-comments/no-use\": \"off\"\n    }\n}\n\n\n6\u3064\u306e\u30eb\u30fc\u30eb\u304c\u3042\u308a\u307e\u3059\u3002\n\neslint-comments/disable-enable-pair\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\n\n\u30eb\u30fc\u30eb\u3092\u4e00\u6642\u7684\u306b\u7121\u52b9\u306b\u3059\u308b /*eslint-disable */\n\n\u4e00\u6642\u7684\u306b\u7121\u52b9\u5316\u3055\u308c\u305f\u30eb\u30fc\u30eb\u3092\u518d\u5ea6\u6709\u52b9\u306b\u3059\u308b /*eslint-enable */\n\n\n\u3092\u5fc5\u305a\u30da\u30a2\u3067\u5229\u7528\u3057\u306a\u3055\u3044\u3001\u3068\u3044\u3046\u30eb\u30fc\u30eb\u3067\u3059\u3002\n\u30da\u30a2\u3068\u306a\u308b /*eslint-enable */ \u30b3\u30e1\u30f3\u30c8\u304c\u5b58\u5728\u3057\u306a\u304b\u3063\u305f /*eslint-disable */ \u30b3\u30e1\u30f3\u30c8\u3092\u8b66\u544a\u3057\u307e\u3059\u3002\n Examples of incorrect code for this rule:\n/*eslint-disable no-console */\n\"\u203bERROR: Requires 'eslint-enable' directive for 'no-console'.\"\n\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n\n Examples of correct code for this rule:\ndoSomething().catch(err => {\n    /*eslint-disable no-console */\n    console.error(err.stack)\n    process.exitCode = 1\n    /*eslint-enable no-console */\n})\n\n\neslint-comments/no-duplicate-disable\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\u65e2\u306b\u4e00\u6642\u7684\u306b\u7121\u52b9\u5316\u3055\u308c\u3066\u3044\u308b\u30eb\u30fc\u30eb\u3092\u3001\u91cd\u306d\u3066\u7121\u52b9\u5316\u3057\u3088\u3046\u3068\u3057\u305f\u6642\u306b\u8b66\u544a\u3057\u307e\u3059\u3002\n\u30b3\u30e1\u30f3\u30c8\u3092\u524a\u9664\u3057\u3066\u7121\u52b9\u5316\u89e3\u9664\u3057\u305f\u3064\u3082\u308a\u304c\u3001\u5225\u306e\u5834\u6240\u306b\u3082\u540c\u3058\u7121\u52b9\u5316\u30b3\u30e1\u30f3\u30c8\u304c\u3042\u3063\u305f\u305f\u3081\u306b\u8b66\u544a\u304c\u5fa9\u6d3b\u3057\u306a\u304b\u3063\u305f... \u3068\u3044\u3046\u3088\u3046\u306a\u30df\u30b9\u3092\u9632\u3052\u307e\u3059\u3002\n Examples of incorrect code for this rule:\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n                             \"\u203bERROR: 'no-console' rule has been disabled already.\"\n    process.exitCode = 1\n})\n\n Examples of correct code for this rule:\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n    process.exitCode = 1\n})\n\n\neslint-comments/no-unlimited-disable\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\u7121\u52b9\u5316\u30b3\u30e1\u30f3\u30c8\u304c\u3042\u3089\u3086\u308b\u30eb\u30fc\u30eb\u3092\u7121\u52b9\u5316\u3057\u3066\u3057\u307e\u3046\u6642\u306b\u8b66\u544a\u3057\u307e\u3059\u3002\n\u30eb\u30fc\u30eb\u540d\u3092\u6307\u5b9a\u305b\u305a\u306b /*eslint-disable */ \u7b49\u306e\u7121\u52b9\u5316\u30b3\u30e1\u30f3\u30c8\u3092\u66f8\u304f\u3068\u3001\u5168\u3066\u306e\u8b66\u544a\u3092\u6291\u5236\u3057\u3066\u3057\u307e\u3044\u307e\u3059\u3002\n\u30eb\u30fc\u30eb\u540d\u3092\u6307\u5b9a\u3057\u3066\u3001\u90aa\u9b54\u306a\u30eb\u30fc\u30eb\u3060\u3051\u3092\u6291\u5236\u3057\u307e\u3057\u3087\u3046\u3002\n Examples of incorrect code for this rule:\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line\n                             \"\u203bERROR: Unexpected unlimited 'eslint-disable-line' comment. Specify some rule names to disable.\"\n    process.exitCode = 1\n})\n\n Examples of correct code for this rule:\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n    process.exitCode = 1\n})\n\n\neslint-comments/no-unused-disable\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\u4e00\u6642\u7684\u306b\u7121\u52b9\u5316\u3057\u305f\u30eb\u30fc\u30eb\u304c\u4e00\u5207\u5831\u544a\u3055\u308c\u306a\u304b\u3063\u305f\u5834\u5408\u306b\u8b66\u544a\u3057\u307e\u3059\u3002\n\u305d\u306e\u3088\u3046\u306a\u30b3\u30e1\u30f3\u30c8\u306f\u5358\u7d14\u306b\u30df\u30b9\u3067\u3042\u308b\u53ef\u80fd\u6027\u304c\u9ad8\u3044\u3067\u3059\u3002\n\u653e\u7f6e\u3059\u308b\u3068\u3001\u5c06\u6765 \u6709\u7528\u306a\u8b66\u544a\u3092\u898b\u9003\u3057\u3066\u3057\u307e\u3046\u304b\u3082\u3057\u308c\u307e\u305b\u3093\u3002\n Examples of incorrect code for this rule:\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-process-exit\n                             \"\u203bERROR: 'no-process-exit' rule is disabled but never reported.\"\n    process.exitCode = 1\n})\n\n\u30bf\u30a4\u30dd\u3082\u5831\u544a\u3055\u308c\u307e\u3059\u3002\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-consola\n                             \"\u203bERROR: 'no-consola' rule is disabled but never reported.\"\n    process.exitCode = 1\n})\n\n Examples of correct code for this rule:\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n    process.exitCode = 1\n})\n\n\neslint-comments/no-unused-enable\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\u4e00\u6642\u7684\u306b\u7121\u52b9\u5316\u3055\u308c\u3066\u3044\u306a\u3044\u30eb\u30fc\u30eb\u3092\u3001\u518d\u5ea6\u6709\u52b9\u5316\u3057\u3088\u3046\u3068\u3057\u305f\u5834\u5408\u306b\u8b66\u544a\u3057\u307e\u3059\u3002\n\u305d\u306e\u3088\u3046\u306a\u6709\u52b9\u5316\u30b3\u30e1\u30f3\u30c8\u306f\u30df\u30b9\u3067\u3042\u308b\u53ef\u80fd\u6027\u304c\u9ad8\u3044\u3067\u3059\u3002\n Examples of incorrect code for this rule:\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n\n/*eslint-enable no-consoll */\n\"\u203bERROR: 'no-consoll' rule is re-enabled but it has not been disabled.\"\n\n Examples of correct code for this rule:\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n\n/*eslint-enable no-console */\n\n\neslint-comments/no-use\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\u305d\u3082\u305d\u3082\u30c7\u30a3\u30ec\u30af\u30c6\u30a3\u30d6 \u30b3\u30e1\u30f3\u30c8\u3092\u7981\u6b62\u3059\u308b\u30eb\u30fc\u30eb\u3067\u3059\u3002\n\u3059\u3079\u3066\u306e\u30c7\u30a3\u30ec\u30af\u30c6\u30a3\u30d6 \u30b3\u30e1\u30f3\u30c8\u3092\u8b66\u544a\u3057\u307e\u3059\u3002\n Examples of incorrect code for this rule:\n/*eslint-disable no-console */\n\"\u203bERROR: Unexpected ESLint directive comment.\"\n\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n\n/*eslint-enable no-console */\n\"\u203bERROR: Unexpected ESLint directive comment.\"\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n                             \"\u203bERROR: Unexpected ESLint directive comment.\"\n    process.exitCode = 1\n})\n\n Examples of correct code for this rule:\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n\n allow\u30aa\u30d7\u30b7\u30e7\u30f3\u306b\u3088\u3063\u3066\u3001\u7279\u5b9a\u306e\u30c7\u30a3\u30ec\u30af\u30c6\u30a3\u30d6 \u30b3\u30e1\u30f3\u30c8\u3060\u3051\u3092\u8a31\u53ef\u3059\u308b\u3053\u3068\u3082\u3067\u304d\u307e\u3059\u3002\n Examples of incorrect code for the {\"allow\": [\"eslint-disable-line\"]} option:\n/*eslint-disable no-console */\n\"\u203bERROR: Unexpected ESLint directive comment.\"\n\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n\n/*eslint-enable no-console */\n\"\u203bERROR: Unexpected ESLint directive comment.\"\n\n Examples of correct code for the {\"allow\": [\"eslint-disable-line\"]} option:\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n    process.exitCode = 1\n})\n\n\n \u65e2\u77e5\u306e\u5236\u9650\u4e8b\u9805\n\u3053\u308c\u3089\u306f ESLint \u30eb\u30fc\u30eb\u3068\u3057\u3066\u63d0\u4f9b\u3055\u308c\u3066\u3044\u308b\u305f\u3081\u3001\u7121\u52b9\u5316\u3059\u308b\u3053\u3068\u304c\u53ef\u80fd\u3067\u3059 (\u3048\u30fc)\u3002\n/*eslint eslint-comments/no-use: off */\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n    process.exitCode = 1\n})\n\n[eslint-plugin-eslint-comments] \u306f\u3001[ESLint] \u30c7\u30a3\u30ec\u30af\u30c6\u30a3\u30d6 \u30b3\u30e1\u30f3\u30c8 (`/*eslint-disable*/`\u306e\u3088\u3046\u306a\u3084\u3064) \u3092\u4f7f\u3046\u3068\u304d\u306e\u30d9\u30b9\u30c8\u30d7\u30e9\u30af\u30c6\u30a3\u30b9\u3092\u307e\u3068\u3081\u305f [ESLint] \u306e\u8ffd\u52a0\u30eb\u30fc\u30eb\u96c6\u3067\u3059\u3002\n\n[ESLint] \u30c7\u30a3\u30ec\u30af\u30c6\u30a3\u30d6 \u30b3\u30e1\u30f3\u30c8\u3092\u4f7f\u3046\u3068\u304d\u306f\u3001\u610f\u56f3\u305b\u305a\u6709\u7528\u306a\u8b66\u544a\u304c\u6291\u5236\u3055\u308c\u3066\u3057\u307e\u3046\u3053\u3068\u3092\u9632\u3050\u305f\u3081\u3001\u53ef\u80fd\u306a\u9650\u308a\u5f71\u97ff\u7bc4\u56f2\u304c\u72ed\u3044\u3088\u3046\u306b\u8a18\u8ff0\u3059\u3079\u304d\u3067\u3059\u3002\n\u3053\u306e\u30d7\u30e9\u30b0\u30a4\u30f3\u306e\u30eb\u30fc\u30eb\u306f\u3001\u305d\u306e\u305f\u3081\u306e\u624b\u52a9\u3051\u3092\u3057\u3066\u304f\u308c\u307e\u3059\u3002\n\n# :cd: \u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\n\n[ESLint] \u30d7\u30e9\u30b0\u30a4\u30f3\u3067\u3059\u306e\u3067\u3001[npm] \u3092\u5229\u7528\u3057\u3066\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u307e\u3059\u3002\n[ESLint] \u3092\u30b0\u30ed\u30fc\u30d0\u30eb (`--global`) \u306b\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u3066\u3044\u308b\u5834\u5408\u306f\u3001\u540c\u3058\u304f\u30b0\u30ed\u30fc\u30d0\u30eb\u3078\u3002[ESLint] \u3092\u30ed\u30fc\u30ab\u30eb (`--save-dev`) \u306b\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u3066\u3044\u308b\u5834\u5408\u306f\u3001\u540c\u3058\u304f\u30ed\u30fc\u30ab\u30eb\u3078\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u3066\u304f\u3060\u3055\u3044\u3002\n\n```bash:Shell\n$ npm install --save-dev eslint eslint-plugin-node\n```\n\n- Node.js 4.0.0 \u4ee5\u964d\u304c\u5fc5\u8981\u3067\u3059\u3002\n- ESLint 3.1.0 \u4ee5\u964d\u304c\u5fc5\u8981\u3067\u3059\u3002\n\n\u306a\u304a\u3001[\u63a8\u5968\u306f\u30ed\u30fc\u30ab\u30eb \u30a4\u30f3\u30b9\u30c8\u30fc\u30eb](http://qiita.com/mysticatea/items/6bd56ff691d3a1577321)\u3067\u3059\u3002\n\n# :book: \u4f7f\u3044\u65b9\n\n\u8a2d\u5b9a\u30d5\u30a1\u30a4\u30eb (`.eslintrc.json`\u7b49) \u306b\u30d7\u30e9\u30b0\u30a4\u30f3\u3068\u30eb\u30fc\u30eb\u306e\u8a2d\u5b9a\u3092\u8ffd\u52a0\u3057\u307e\u3059\u3002\n\n```json:\u4f8b\n{\n    \"extends\": [\"eslint:recommended\"],\n    \"plugins\": [\"eslint-comments\"],\n    \"rules\": {\n        \"eslint-comments/disable-enable-pair\": \"error\",\n        \"eslint-comments/no-duplicate-disable\": \"error\",\n        \"eslint-comments/no-unlimited-disable\": \"error\",\n        \"eslint-comments/no-unused-disable\": \"error\",\n        \"eslint-comments/no-unused-enable\": \"error\",\n        \"eslint-comments/no-use\": \"off\"\n    }\n}\n```\n\n6\u3064\u306e\u30eb\u30fc\u30eb\u304c\u3042\u308a\u307e\u3059\u3002\n\n## [eslint-comments/disable-enable-pair]\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\n\n- \u30eb\u30fc\u30eb\u3092\u4e00\u6642\u7684\u306b\u7121\u52b9\u306b\u3059\u308b `/*eslint-disable */`\n- \u4e00\u6642\u7684\u306b\u7121\u52b9\u5316\u3055\u308c\u305f\u30eb\u30fc\u30eb\u3092\u518d\u5ea6\u6709\u52b9\u306b\u3059\u308b `/*eslint-enable */`\n\n\u3092\u5fc5\u305a\u30da\u30a2\u3067\u5229\u7528\u3057\u306a\u3055\u3044\u3001\u3068\u3044\u3046\u30eb\u30fc\u30eb\u3067\u3059\u3002\n\u30da\u30a2\u3068\u306a\u308b `/*eslint-enable */` \u30b3\u30e1\u30f3\u30c8\u304c\u5b58\u5728\u3057\u306a\u304b\u3063\u305f `/*eslint-disable */` \u30b3\u30e1\u30f3\u30c8\u3092\u8b66\u544a\u3057\u307e\u3059\u3002\n\n:-1: Examples of **incorrect** code for this rule:\n\n```js\n/*eslint-disable no-console */\n\"\u203bERROR: Requires 'eslint-enable' directive for 'no-console'.\"\n\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n```\n\n:+1: Examples of **correct** code for this rule:\n\n```js\ndoSomething().catch(err => {\n    /*eslint-disable no-console */\n    console.error(err.stack)\n    process.exitCode = 1\n    /*eslint-enable no-console */\n})\n```\n\n## [eslint-comments/no-duplicate-disable]\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\u65e2\u306b\u4e00\u6642\u7684\u306b\u7121\u52b9\u5316\u3055\u308c\u3066\u3044\u308b\u30eb\u30fc\u30eb\u3092\u3001\u91cd\u306d\u3066\u7121\u52b9\u5316\u3057\u3088\u3046\u3068\u3057\u305f\u6642\u306b\u8b66\u544a\u3057\u307e\u3059\u3002\n\u30b3\u30e1\u30f3\u30c8\u3092\u524a\u9664\u3057\u3066\u7121\u52b9\u5316\u89e3\u9664\u3057\u305f\u3064\u3082\u308a\u304c\u3001\u5225\u306e\u5834\u6240\u306b\u3082\u540c\u3058\u7121\u52b9\u5316\u30b3\u30e1\u30f3\u30c8\u304c\u3042\u3063\u305f\u305f\u3081\u306b\u8b66\u544a\u304c\u5fa9\u6d3b\u3057\u306a\u304b\u3063\u305f... \u3068\u3044\u3046\u3088\u3046\u306a\u30df\u30b9\u3092\u9632\u3052\u307e\u3059\u3002\n\n:-1: Examples of **incorrect** code for this rule:\n\n```js\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n                             \"\u203bERROR: 'no-console' rule has been disabled already.\"\n    process.exitCode = 1\n})\n```\n\n:+1: Examples of **correct** code for this rule:\n\n```js\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n    process.exitCode = 1\n})\n```\n\n## [eslint-comments/no-unlimited-disable]\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\u7121\u52b9\u5316\u30b3\u30e1\u30f3\u30c8\u304c\u3042\u3089\u3086\u308b\u30eb\u30fc\u30eb\u3092\u7121\u52b9\u5316\u3057\u3066\u3057\u307e\u3046\u6642\u306b\u8b66\u544a\u3057\u307e\u3059\u3002\n\u30eb\u30fc\u30eb\u540d\u3092\u6307\u5b9a\u305b\u305a\u306b `/*eslint-disable */` \u7b49\u306e\u7121\u52b9\u5316\u30b3\u30e1\u30f3\u30c8\u3092\u66f8\u304f\u3068\u3001\u5168\u3066\u306e\u8b66\u544a\u3092\u6291\u5236\u3057\u3066\u3057\u307e\u3044\u307e\u3059\u3002\n\u30eb\u30fc\u30eb\u540d\u3092\u6307\u5b9a\u3057\u3066\u3001\u90aa\u9b54\u306a\u30eb\u30fc\u30eb\u3060\u3051\u3092\u6291\u5236\u3057\u307e\u3057\u3087\u3046\u3002\n\n:-1: Examples of **incorrect** code for this rule:\n\n```js\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line\n                             \"\u203bERROR: Unexpected unlimited 'eslint-disable-line' comment. Specify some rule names to disable.\"\n    process.exitCode = 1\n})\n```\n\n:+1: Examples of **correct** code for this rule:\n\n```js\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n    process.exitCode = 1\n})\n```\n\n## [eslint-comments/no-unused-disable]\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\u4e00\u6642\u7684\u306b\u7121\u52b9\u5316\u3057\u305f\u30eb\u30fc\u30eb\u304c\u4e00\u5207\u5831\u544a\u3055\u308c\u306a\u304b\u3063\u305f\u5834\u5408\u306b\u8b66\u544a\u3057\u307e\u3059\u3002\n\u305d\u306e\u3088\u3046\u306a\u30b3\u30e1\u30f3\u30c8\u306f\u5358\u7d14\u306b\u30df\u30b9\u3067\u3042\u308b\u53ef\u80fd\u6027\u304c\u9ad8\u3044\u3067\u3059\u3002\n\u653e\u7f6e\u3059\u308b\u3068\u3001\u5c06\u6765 \u6709\u7528\u306a\u8b66\u544a\u3092\u898b\u9003\u3057\u3066\u3057\u307e\u3046\u304b\u3082\u3057\u308c\u307e\u305b\u3093\u3002\n\n:-1: Examples of **incorrect** code for this rule:\n\n```js\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-process-exit\n                             \"\u203bERROR: 'no-process-exit' rule is disabled but never reported.\"\n    process.exitCode = 1\n})\n```\n\n\u30bf\u30a4\u30dd\u3082\u5831\u544a\u3055\u308c\u307e\u3059\u3002\n\n```js\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-consola\n                             \"\u203bERROR: 'no-consola' rule is disabled but never reported.\"\n    process.exitCode = 1\n})\n```\n\n:+1: Examples of **correct** code for this rule:\n\n```js\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n    process.exitCode = 1\n})\n```\n\n## [eslint-comments/no-unused-enable]\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\u4e00\u6642\u7684\u306b\u7121\u52b9\u5316\u3055\u308c\u3066\u3044\u306a\u3044\u30eb\u30fc\u30eb\u3092\u3001\u518d\u5ea6\u6709\u52b9\u5316\u3057\u3088\u3046\u3068\u3057\u305f\u5834\u5408\u306b\u8b66\u544a\u3057\u307e\u3059\u3002\n\u305d\u306e\u3088\u3046\u306a\u6709\u52b9\u5316\u30b3\u30e1\u30f3\u30c8\u306f\u30df\u30b9\u3067\u3042\u308b\u53ef\u80fd\u6027\u304c\u9ad8\u3044\u3067\u3059\u3002\n\n:-1: Examples of **incorrect** code for this rule:\n\n```js\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n\n/*eslint-enable no-consoll */\n\"\u203bERROR: 'no-consoll' rule is re-enabled but it has not been disabled.\"\n```\n\n:+1: Examples of **correct** code for this rule:\n\n```js\n/*eslint-disable no-console */\n\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n\n/*eslint-enable no-console */\n```\n\n## [eslint-comments/no-use]\n\n\u3053\u306e\u30eb\u30fc\u30eb\u306f\u3001\u305d\u3082\u305d\u3082\u30c7\u30a3\u30ec\u30af\u30c6\u30a3\u30d6 \u30b3\u30e1\u30f3\u30c8\u3092\u7981\u6b62\u3059\u308b\u30eb\u30fc\u30eb\u3067\u3059\u3002\n\u3059\u3079\u3066\u306e\u30c7\u30a3\u30ec\u30af\u30c6\u30a3\u30d6 \u30b3\u30e1\u30f3\u30c8\u3092\u8b66\u544a\u3057\u307e\u3059\u3002\n\n:-1: Examples of **incorrect** code for this rule:\n\n```js\n/*eslint-disable no-console */\n\"\u203bERROR: Unexpected ESLint directive comment.\"\n\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n\n/*eslint-enable no-console */\n\"\u203bERROR: Unexpected ESLint directive comment.\"\n```\n\n```js\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n                             \"\u203bERROR: Unexpected ESLint directive comment.\"\n    process.exitCode = 1\n})\n```\n\n:+1: Examples of **correct** code for this rule:\n\n```js\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n```\n\n:wrench: `allow`\u30aa\u30d7\u30b7\u30e7\u30f3\u306b\u3088\u3063\u3066\u3001\u7279\u5b9a\u306e\u30c7\u30a3\u30ec\u30af\u30c6\u30a3\u30d6 \u30b3\u30e1\u30f3\u30c8\u3060\u3051\u3092\u8a31\u53ef\u3059\u308b\u3053\u3068\u3082\u3067\u304d\u307e\u3059\u3002\n\n:-1: Examples of **incorrect** code for the `{\"allow\": [\"eslint-disable-line\"]}` option:\n\n```js\n/*eslint-disable no-console */\n\"\u203bERROR: Unexpected ESLint directive comment.\"\n\ndoSomething().catch(err => {\n    console.error(err.stack)\n    process.exitCode = 1\n})\n\n/*eslint-enable no-console */\n\"\u203bERROR: Unexpected ESLint directive comment.\"\n```\n\n:+1: Examples of **correct** code for the `{\"allow\": [\"eslint-disable-line\"]}` option:\n\n```js\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n    process.exitCode = 1\n})\n```\n\n# :warning: \u65e2\u77e5\u306e\u5236\u9650\u4e8b\u9805\n\n\u3053\u308c\u3089\u306f ESLint \u30eb\u30fc\u30eb\u3068\u3057\u3066\u63d0\u4f9b\u3055\u308c\u3066\u3044\u308b\u305f\u3081\u3001\u7121\u52b9\u5316\u3059\u308b\u3053\u3068\u304c\u53ef\u80fd\u3067\u3059 (\u3048\u30fc)\u3002\n\n```js\n/*eslint eslint-comments/no-use: off */\n\ndoSomething().catch(err => {\n    console.error(err.stack) //eslint-disable-line no-console\n    process.exitCode = 1\n})\n```\n\n[ESLint]: http://eslint.org/\n[eslint-plugin-eslint-comments]: https://github.com/mysticatea/eslint-plugin-eslint-comments#eslint-plugin-eslint-comments\n[eslint-comments/disable-enable-pair]: https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/disable-enable-pair.md\n[eslint-comments/no-duplicate-disable]: https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/no-duplicate-disable.md\n[eslint-comments/no-unlimited-disable]: https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/no-unlimited-disable.md\n[eslint-comments/no-unused-disable]: https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/no-unused-disable.md\n[eslint-comments/no-unused-enable]: https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/no-unused-enable.md\n[eslint-comments/no-use]: https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/no-use.md\n[npm]: https://www.npmjs.com/\n", "tags": ["JavaScript", "ESLint", "Node.js"]}